app/models/interval_receiver.rb

```diff
@@ -1,10 +1,24 @@
 # frozen_string_literal: true
 
 class IntervalReceiver < EventReceiver
+  # Callbacks
+  after_create :create_internal_job!
+
+  # Validations
   validates :start_time, :interval, presence: true
 
-  # Only called from sidekiq job, at certain intervals
+  # Sidekiq job scheduled after_create => trigger condition will always be met
   def trigger_condition_met?(_job)
     true
   end
+
+  def next_interval_time
+    (((Time.zone.now.to_i - start_time) / interval).ceil + interval.seconds)
```

Comment:
Should this be `(((Time.zone.now.to_i - start_time) / interval).ceil * interval).seconds`?